### ISSUE TITLE
Check if data contains inf values

### ISSUE THREAD

**Issue by phackstock** [IAMconsortium/nomenclature#387]
We could implement a validator that checks if a given data set contains any inf values.

**Comment by danielhuppmann** [comment]
For my understanding, what's the rationale? Because int-values can't be saved to a database?

**Comment by phackstock** [comment]
> Because int-values can't be saved to a database?

Correct, ORACLE throws an error if you're trying to save the numeric value `inf`.
Every now and then a user tries to submit data containing inf values. It's always been a mistake on the data side but the ORACLE error is not super useful.

**Comment by danielhuppmann** [comment]
This issue is now becoming relevant again because **ixmp4** can handle int-values when writing directly to the database but this causes issues with the API (and downstream services like visualizations in app).

https://github.com/iiasa/ixmp4/issues/215

Please implement a check as part of `DataStructureDefinition.validate()` to always check for inf and -inf values and raise an error accordingly.
